src/ast/types/ast.ts

```diff
@@ -47,5 +47,5 @@ export interface Location {
 
 export interface BaseNode {
   kind: string;
-  location: Location;
+  location?: Location;
```

Comment:
I thought making `location` optional might be useful because in cases where we are creating the node ourselves, the `location` information is unavailable (e.g., when creating an AST independent from the parser for testing purposes, some functions in `ec-evaluator/nodeCreator.ts`). Of course adding in crafted location information is also fine, just suggesting an alternative here. Let me know if you prefer it to be non-optional.

Reply:
Ah I see, I think making `location` optional is justifiable then.